src/js/index.spec.js

```diff
@@ -0,0 +1,47 @@
+import '@testing-library/jest-dom';
+import { renderJoke } from './renderJoke';
+import { showInitialJoke } from './index.js';
+import { bindEventListenerToRefreshButton } from './bindEventListenerToRefreshButton';
+import { jokeFetcher, jokeFetcherUpperBounds } from './jokeFetcher';
+
+jest.mock('./renderJoke', () => ({
+  ...jest.requireActual('./renderJoke'),
+  renderJoke: jest.fn(),
+}));
+
+jest.mock('./bindEventListenerToRefreshButton', () => ({
+  ...jest.requireActual('./bindEventListenerToRefreshButton'),
+  bindEventListenerToRefreshButton: jest.fn(),
+}));
+
+jest.mock('./jokeFetcher', () => ({
+  ...jest.requireActual('./jokeFetcher'),
+  jokeFetcher: jest.fn(),
+  jokeFetcherUpperBounds: jest.fn(),
+}));
+
+describe('showInitialJoke', () => {
+  it('calls jokeFetcherUpperBounds', () => {
+    showInitialJoke();
+
+    expect(jokeFetcherUpperBounds).toHaveBeenCalled();
+  });
+
+  it('calls jokeFetcher', () => {
+    showInitialJoke();
```

Comment:
Something kind of interesting here is that we don't technically have to call `showInitialJoke()` again for `jokeFetcher()`, `renderJoke()`, and `bindEventListenerToRefreshButton()` to be called. We haven't done anything to [tear down](https://jestjs.io/docs/setup-teardown)/clear this test suite, so calls made from the initial call are still there.

I did a bit of poking around here, and it turns out that if you comment out each instance of `showInitialJoke()` (including the very first instance), all of these tests still pass. I think what's happening is that these individual functions (e.g. `jokeFetcherUpperBounds()`, `renderJoke()`, etc.) are being called in other test suites. These calls are never cleared or reset, so even if `showInitialJoke()` isn't actually working, it still appears like all of these functions are being called.

I think you'll need to think about doing something like [clearing mocks](https://jestjs.io/docs/jest-object#jestclearallmocks) for these tests to really have value.

Side note: if you run `.toHaveBeenCalled(1)` without changing anything else, you can see that these functions are actually being called multiple times.